util/sym-mode_test.go

```diff
@@ -0,0 +1,58 @@
+// Mgmt
+// Copyright (C) 2013-2019+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package util_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/purpleidea/mgmt/util"
+)
+
+func TestSymbolicMode(t *testing.T) {
+	def := os.FileMode(0644) | os.ModeSetgid
+	symModeTests := []struct {
+		input  string
+		expect os.FileMode
```

Comment:
Please look at our other tests to add an `experr` sort of variable or similar. The recent pr by @dbakong shows this too.

Reply:
In that PR the errors are being compared and even though  the text is the same they are separate error values so it won't pass that way ([playground](https://play.golang.org/p/nEYvEGxgVg3)). Would it be preferred to check for nil then compare output of `Error()` or could error constants and `errors.Is` be used?

Reply:
If we made that error compare somewhere, please let me know! In general it's best to compare error constants, or if it's part of a test case and we're checking for a particular error that the user never needs to compare to, then the string compare is okay.